SU-SWS/decanter
# Refactor card component using placeholder and match Figma design

# READY FOR REVIEW

# Summary
- Refactor the card component to use new modular-spacing mixin and make sure it looks the same as the mocks in Figma. Buttons in cards weren't showing up but it should now.
- ~~Create card mixin to allow passing in options - orientation, drop shadow depth, border,  headline font family and vertical alignment of content (for horizontal variant)~~
- Use %placeholder and @extend to reduce repetitive code
- Set some max-width for components (cards, CTAs) in the kss scss files so devs/designers visiting the site can get a better idea of what the components look like on a real site.

# Needed By (Date)
- Whenever

# Urgency
- Not urgent

# Steps to Test

1. Pull this branch
2. Run `grunt styleguide`
3. Check the styleguide that the default renders for both vertical and horizontal cards match the mocks in Figma. Check that the Hero component also looks like (it uses the card component).

# Affected Projects or Products
- Decanter
- Event calendar

# Associated Issues and/or People
- Talked to @JohnHolleman about default CTA link and button styles (in general, not necessarily the ones inside the cards). He said CTA links should be semibold (font-weight: 600, which is actually same as variable $font-bold in Decanter). On the other hand, font-weight for text in buttons should be normal (font-weight: 400). Both are used in cards as well so I made the tweak in the Button and Link component.
- Due to the inherent line-height of our font stack, I had to reduce some of the modular-spacing steps (compared to what Figma specified) to match the spacing of the rendered cards and the mocks visually. I talked to @JohnHolleman about this and he said he would rather we match the design visually than having the modular step numbers match exactly.
- I took away the margins for the Button component so it's consistent with other components (no spacing on the outside of the component).
